### DISCUSSION TITLE
.net6 para .net5

### DISCUSSION THREAD

**Discussion by 123Will** [balta-io/2801#85] [Perguntas e Respostas] [answered]
No começo do curso instalei o .net6, porem na hora que abre o Program.cs não aparece o namespace, internal class e etc...
então por estar divergente desinstalei o .net6 e coloquei o .net5 e ai deu certo, mas na hora de por o comando dotnet run aparece esse erro a seguir.
PS C:\Users\Willian\calculator> dotnet run
C:\Program Files\dotnet\sdk\5.0.214\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.TargetFrameworkInference.targets(141,5): error NETSDK1045: O SDK do .NET atual não dá suporte para direcionar a .NET 6.0. Direcione a .NET 5.0 ou inferior, ou use uma versão do SDK do .NET compatível com  .NET 6.0. [C:\Users\Willian\calculator\Calculator.csproj]

Ocorreu uma falha no build. Corrija os erros de build e execute novamente.

oque eu devo fazer para corrigir e ficar apenas com o .net5. Obrigado.

**Comment by victorpomim** [comment, marked as answer]
Fala 123Will, beleza? 
Cara, eu estou fazendo o curso direto com o .net6, você não precisa se preocupar, a atualização realizada apenas "limpou" um pouco alguns códigos desnecessários, o que não interfere na apresentação dos conteúdos em geral. 
Você pode inserir os comandos manualmente, e continuar a utilizar os modelos feitos pelo Balta. 

Por ex. 
Agora não é necessário a importação do: using System, entre outras bibliotecas, são considerados como códigos mortos pelo VSCode. 

Espero que tenha sanado um pouco sua dúvida, mas caso queira utilizar o .net antigo, segue um link da Microsoft: 
https://docs.microsoft.com/en-us/dotnet/core/additional-tools/uninstall-tool?tabs=windows

**Reply by 123Will** [reply]
fala victor, no curso aparece ele usando a versão 3, eu baixei a 6, eu segui todos os passos que ele foi passando, mas na hora que chega na parte de jogar a pasta no studio code e abri o (program.csproj), ele aparece apenas uma linha escrito (Hello World) e no curso aparece todo o scopo com
     namespace e etc..

**Reply by 123Will** [reply]
.net6
console.writeline("Hello World");

.net5 e inferiores
Using System

Namespace MeuApp

    class Program ..... e por ai vai

a versão 6 aparece diferente do curso ai nao consigo mais por o códigos sempre da erro

**Reply by victorpomim** [reply]
Entendi. 
Isso não é um "erro" de fato, é só uma otimização do layout do .NET. 
Se você fizer a inserção destes dados manualmente, você vai ver que não gerará nenhum erro no código **apenas a mensagem de diretiva de uso desnecessária**. 
Veja a mensagem que aparece ao realizar a Inserção do using System; manualmente: 

![image](https://user-images.githubusercontent.com/84252190/175766862-d51399e5-e0b0-45a0-a0ce-ad66f78439aa.png)

Agora, eu vou retirar a inserção do using System, e você vai ver que não gerará erro: 

![image](https://user-images.githubusercontent.com/84252190/175766980-bbd59116-7e66-4ecd-9980-b53ab77a415d.png)

Eu atualizei um projeto do balta.io [Editor HTML] para ficar mais fácil a sua visualização/entendimento. 
Agora ao invés de vir pronto, você pode realizar a inserção manual do namespace, e criar a class Program, que não é para gerar erro! 

Segue como ficará o código do Editor HTML (essa é a pasta do Editor.cs):

```
`using System.Text;

namespace EditorHtml {
    public static class Editor {
        public static void Show(){
            Console.Clear();
            Console.BackgroundColor = ConsoleColor.White;
            Console.ForegroundColor = ConsoleColor.Black;
            Console.Clear();
            Console.WriteLine("EDITOR MODE");
            Console.WriteLine("-----------");
        }


```        public static void Start() {
            var  file = new StringBuilder();

            do { 
                file.Append(Console.ReadLine());
                file.Append(Environment.NewLine);
            }   while(Console.ReadKey().Key != ConsoleKey.Escape);

            Console.WriteLine("-----------");
            Console.WriteLine("Do you wanna save the file?");
            Console.ReadLine();

        }
    }

}`

**Reply by 123Will** [reply]
Bom dia, eu baixei o .net6 novamente e voltei na parte da calculadora pra refazer a aula, mas deu 3 erros... 1  error CS1529: 2  error CS1056:
3  error CS1513:... não consigo mandar os códigos mais, to apanhando muito ja faz 10 dias que não consigo sair dessa calculadora cada dia é um erro diferente rsrs, to quase desistindo desse curso. Obrigado pela atenção.

**Reply by 123Will** [reply]
![Sem título](https://user-images.githubusercontent.com/97917861/175817501-ba7ce8ff-e2dd-4151-a549-f91eab0267c5.png)

**Reply by victorpomim** [reply]
Fala Will, beleza? 
Cara, não desiste não!! Foco e força na peruca! 
Vou te ensinar uma coisa aqui, que caso você queira seguir na carreira, vai ser muito comum na sua vida! 
Não entenda como crítica, é um conselho de um amigo que também está começando neste mundo da programação, mas teve alguma orientação e aprendeu com alguns colegas já no ramo.
Sempre que tiver com algum problema em relação aos erros, você precisa ir na documentação da nossa linguagem (No caso o C#, para pesquisar sobre o erro e caso não consiga resolver, vir para os fóruns, stackoverflow e afins). 

De toda forma, eu gosto de ajudar porque sua dúvida pode ter sido a minha e eventualmente, eu posso reafirmar meus ensinamentos ao explicar/achar a solução pra você! 

O primeiro erro que você me apresentou é o CS1529 - a documentação nos traz a resposta para isso - https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1529
Mas vou te explicar aqui, o using system, tem que estar acima de qualquer classe e ou indicação. Ali você está pedindo para o compilador executar um Console.WriteLine e depois, o programa.
Começa apagando essa linha 1 e 2. 

O segundo erro CS1056 temos a resposta na documentação neste link: https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1056
Neste caso, basicamente é que na hora da execução está conflitando com algum caractere especial na hora de rodar o código, por aqui pelo print apresentado eu acho que é essa aspas simples antes do - **'**using System.Text;- deixei em negrito o que eu iria corrigir no código, pode ser que esteja no final do seu código também, verifica isso.

Já no ultimo erro o CS1513 - temos o link da resposta na documentação - https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1513
Basicamente, teve alguma chave que você não fechou no seu código, então precisa revisar se tudo o que tu abriu você fechou! 
Faça essas correções, e veja no que dá! 

Em último caso, manda aqui de novo, que eu te passo o meu código utilizado na Calculadora para vc comparar e testar, demoro??

Ademais, nessa aula nós não usamos o using Sytem.text; 
o System.Text é utilizado para manipulação de String e etc, que você vai ver mais pra frente com StringBuilder e etc, então muda isso também. 

Acredito que tudo isso possa te ajudar e continuar no curso! 
Abraço!! 
Bom final de semana!

**Reply by andrebaltieri** [reply]
@123Will não desiste não!!! começo é assim mesmo... passa as dificuldades aí que ajudamos... tem nosso Discord (https://balta.io/discord) e nossa central de ajuda (https://balta.io/ajuda) se precisar.... 💜
